packages/terra-button/tests/jest/Button.test.jsx

```diff
@@ -225,4 +227,13 @@ it('should render a button with accessible label', () => {
   const button = shallow(<Button text="Button" aria-label="Custom Accessible Label" />);
   expect(button).toMatchSnapshot();
 });
+
+it('correctly applies the theme context className', () => {
+  const button = mount(
+    <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
+      <Button text="neutral" />
+    </ThemeContextProvider>,
+  );
+  expect(button).toMatchSnapshot();
+});
```

Comment:
Just a note here: I didn't like how I ended up having to write this test. The context api's for enzyme did not work, mocking was taking too long to figure out and shallow rendering still did not apply the context. 
I don't want consumers pulling in the terra-theme-context package directly to get access to the provider... If people have to test this way, we should provide the provider from the terra-application package.

Reply:
I'm pretty sure the enzyme context API only works when you set context like:
```
Component.context = Context;
```
I had my own slew of troubles getting the mocking to work for terra-action-header and opted to use `React.useContext` so I could mock like:
```
jest.spyOn(React, 'useContext).mockReturnValue({ className: 'orion-fusion-theme' });
```

Reply:
+1 to exposing a theme provider in terra-application

Reply:
Enzyme's context APIs for shallow/mount use the old context API, which we're avoiding completely. This is annoying here, but the uplift to react-intl v4 is going to be much more disruptive (no more shallow/mountWithIntl :S).